### PR TITLE
Tensorflow frontend tensor sub instance method

### LINKED ISSUE #4833: sub

### PR DESCRIPTION
**Pull request by khimraj**
close #4833

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_tensor.py [hunk 1]
```diff
@@ -118,3 +118,36 @@ def test_tensorflow_instance_get_shape(
...
+        input_dtypes=input_dtype,
+        as_variable_flags=as_variable,
+        with_out=False,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        fw=fw,
+        frontend="tensorflow",
+        frontend_class=Tensor,
+        fn_tree="Tensor.sub",
+        self=np.asarray(x[0], dtype=input_dtype[0]),
```
Comment (zaeemansari70) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_tensor.py:151]: We have some changes in the policy recently, there is no need to cast the arrays before passing them into the test function now.

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_tensor.py [hunk 2]
```diff
@@ -118,3 +118,36 @@ def test_tensorflow_instance_get_shape(
...
+        y=np.asarray(x[1], dtype=input_dtype[1]),
```
Comment (zaeemansari70) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_tensor.py:152]: Same

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Hey! 
First of all thank you soo much for contributing to ivy! 🙂
Your PR looks amazing ✨!
Just requested a few changes and left comments on them which needs to be addressed before we can merge your PR! 🙂
Thanks!

**Comment by khimraj** [comment]
Hi @zaeemansari70, thanks a lot for reviewing this. I have made requested changes. Please check and approve whenever possible.

File: ivy/functional/frontends/tensorflow/tensor.py [hunk 3]
```diff
@@ -22,3 +22,6 @@ def add(self, y, name="add"):
 
     def get_shape(self):
         return tf_frontend.Shape(input=self.data)
+
+    def sub(self, y, name="sub"):
```
Comment (zaeemansari70) [inline ivy/functional/frontends/tensorflow/tensor.py:26]: Here

File: ivy_tests/test_ivy/test_frontends/test_tensorflow/test_tensor.py [hunk 4]
```diff
@@ -118,3 +118,36 @@ def test_tensorflow_instance_get_shape(
...
+@given(
+    dtype_and_x=helpers.dtype_and_values(
+        available_dtypes=tuple(
+            set(ivy_np.valid_float_dtypes).intersection(set(ivy_tf.valid_float_dtypes))
+        ),
+        num_arrays=2,
+        shared_dtype=True,
+    ),
+    num_positional_args=helpers.num_positional_args(
+        fn_name="ivy.functional.frontends.tensorflow.Tensor.sub",
```
Comment (zaeemansari70) [inline ivy_tests/test_ivy/test_frontends/test_tensorflow/test_tensor.py:134]: here and below where required

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
The name of the function is `subtract` in the tensorflow frontend class, not `sub`, which needs to be fixed wherever used.

File: ivy/functional/frontends/tensorflow/tensor.py (hunk 3 above)
Comment (khimraj) [reply ivy/functional/frontends/tensorflow/tensor.py:26]: [Instance Methods](https://www.tensorflow.org/api_docs/python/tf/Tensor) reference is given  in issue #3616. Here it is mentioned  as `sub`. Can you please reconfirm this whenever possible, I just want to confirm before making these changes.
Thanks!

**Comment by khimraj** [comment]
@zaeemansari70, really sorry for above mess up. Can you please review now whenever you have time.
Thanks!

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
No worries!
Thanks for the changes, just a small change requested!

**Comment by khimraj-ivyhomes** [comment]
@zaeemansari70, I have made highlighted changes. Can you please review it now.

**Review by zaeemansari70 — APPROVED** [review]
lgtm!

**Pull request merged**